terraform/modules/gro_ingestion_service/lambdas.tf

```diff
@@ -0,0 +1,69 @@
+locals {
+  lambda_runtime = "nodejs18.x"
+  functions = toset([
+    for file in fileset("${path.module}/lambdas/src/functions", "*.function.ts") : lower(replace(split(".", split("/", file)[length(split("/", file)) - 1])[0], "[^a-zA-Z0-9]", "-"))
+  ])
+}
+
+data "archive_file" "typescript-source" {
+  type        = "zip"
+  source_dir  = "${path.module}/lambdas/src"
+  output_path = "zip/typescript-source.zip"
+}
+
+data "archive_file" "lambda-function-source" {
+  type        = "zip"
+  source_dir  = "${path.module}/lambdas/dist"
+  output_path = "zip/lambda-function-source.zip"
+  depends_on  = [null_resource.lambda-function-source-builder]
+}
+
+resource "null_resource" "lambda-function-source-builder" {
+  provisioner "local-exec" {
+    working_dir = "${path.module}/lambdas"
+    command     = <<EOT
+      npm i
+      npm run build
+    EOT
+  }
+  triggers = {
+    lambda-function-md5 = data.archive_file.typescript-source.output_md5
```

Comment:
This surprised me that it was triggered by the md5

Reply:
Updated